micro-rdk/src/common/data_manager.rs

```diff
@@ -241,25 +244,40 @@ where
     async fn read_messages_for_collector(
         &self,
         collector_key: &ResourceMethodKey,
-    ) -> Result<Vec<SensorData>, DataSyncError> {
+    ) -> Result<Vec<Vec<SensorData>>, DataSyncError> {
         let store_lock = self.store.lock().await;
-        let mut raw_messages: Vec<BytesMut> = vec![];
+        let mut chunks: Vec<Vec<SensorData>> = vec![];
```

Comment:
Per discussion, I think this isn't currently achieving the goal of keeping the memory usage for each upload under a static limit.